Task2.StringConverterLibrary/StringConverter.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task2.StringConverterLibrary
+{
+    public static class StringConverter
+    {
+        public static int ToInt (this string str)
+        {
+            if (String.IsNullOrWhiteSpace(str))
+            {
+                throw new ArgumentException($"{nameof(str)} is null, empty or whitespace");
+            }
+
+            int result = 0, sign, startPosition;
+
+            startPosition = ProcessFirstSymbol(str[0],out sign);
+
+            try
+            {
+                checked
```

Comment:
Also we can use:
int overflow = int.MaxValue/10;
			    if (number > overflow)
			    {
				    throw new OverflowException();
			    }